ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,576 @@
+
+Notice that our schema already has a JSONB field in :code:`clicks` called :code:`user_data`. Each tenant can use it for flexible storage.
+
+Supposing company 5 uses the field to track the user agent ad clicks, they can later query to find which browsers click most often:
+
+.. code-block:: postgresql
+
+  SELECT
+    user_data->>'browser' AS browser,
+    count(*) AS count
+  FROM clicks
+  WHERE company_id = 5
+  GROUP BY user_data->>'browser'
+  ORDER BY count DESC
+  LIMIT 10;
+
+The database administrator can even create a `partial index <https://www.postgresql.org/docs/current/static/indexes-partial.html>`_ to improve speed for an individual tenant's query patterns. Here is one to improve company 5's filters for browser information:
+
+.. code-block:: postgresql
+
+  CREATE INDEX click_user_data_browser
+  ON clicks (user_data->>'browser')
+  WHERE company_id = 5;
+
+Additionally, PostgreSQL supports GIN indices on JSONB. Creating a GIN index on a JSONB column will create an index on every key and value within that JSON document. This speeds up a number of `JSONB operators <https://www.postgresql.org/docs/current/static/functions-json.html#FUNCTIONS-JSONB-OP-TABLE>`_ such as :code:`?`, :code:`?|`, and :code:`?&`.
+
+.. code-block:: postgresql
+
+  CREATE INDEX click_user_data
+  ON clicks GIN (user_data);
+
+  -- this allows queries like, "which clicks have
+  -- the browser key present in user_data?"
+
+  SELECT id
+    FROM clicks
+   WHERE user_data ? 'browser'
+     AND company_id = 5;
+
+Scaling Hardware Resources
+--------------------------
+
+Multi-tenant databases should be designed for future scale as business grows or tenants want to store more data. Citus can scale out easily by adding new machines without having to make any changes or take application downtime.
+
+Being able to rebalance data in the Citus cluster allows you to grow your data size or number of customers and improve performance on demand. Adding new machines allows you to keep data in memory even when it is much larger than what a single machine can store.
+
+Also, if data increases for only a few large tenants, then you can isolate those particular tenants to separate nodes for better performance.
+
+.. note::
+
+  This section assumes you are using Citus Cloud and have created the schema and loaded data as described earlier. Additionally the Citus Cloud "Dev Plan" will not work because it does not allow scaling nodes; you must use a "Customized Plan."
```

Comment:
That is, I see a few "Predefined Plans" but they seem internal, not sure which ones are available to new users.